### PR TITLE
Loading swap button

### PR DESCRIPTION
**Pull request by wixzi**
# Summary

Fixes Issue #994 #894

Show a loading state till all the exchanges for swap is fetched.

https://user-images.githubusercontent.com/102727631/170865192-7bd2e40a-ec32-446a-9734-e4bf5f45d442.mp4

  # To Test

Go to swap page => Select both tokens => Enter a value => Loading button should show till all the exchanges are loaded.

### CODE REVIEW

**Comment by 0xVenky** [comment]
Hey, a couple of feedback to this:
- Looks very choppy if we can fetch the price from all dexes within a second. Would be good to just let the animation run for 1 to 1.5 seconds even if we find all the prices.
- It would also be good if we just show the dexes that are native to the chain. When I use Gnosis chain, I dont want to see DFYN as it might confuse the user or make the user wonder why we are finding from this dex.
- The loading of images seems to be pretty laggy compared to the text and that makes it look even more choppy. :(

https://user-images.githubusercontent.com/55109759/170916176-38de5e06-fa58-464c-8608-41783d9f0ba2.mov

- We need to make the state change better from "Finding best Price" to "Swap with xyz" Maybe @zamli can provide some thoughts.

**Comment by wixzi** [comment]
> https://deploy-preview-1033--swapr.netlify.app/

Can you share which dexes per network. I added al the dexes we have. When I tested it was min loading 0.5 sec. May be my network and machine is slow. I will try to make it loading for .5sec minimum

For icons, I think we need to preload all network and dex icons. I will do that also. Thanks for the comments.

**Comment by niemam29** [comment]
Hey, I've founded that there are some problems with validating zeros (take a look at the recording). They do not occur on the develop branch

https://user-images.githubusercontent.com/30408272/170954465-b50b69f1-5998-40e1-95ff-492e4b1a5626.mp4

**Comment by MilanVojnovic95** [comment]
Tested and still sometimes a bug appears from [#894](https://github.com/levelkdev/swapr-dapp/issues/894)
and loading animation didn't work this time.


https://user-images.githubusercontent.com/57331244/170965057-f8aaabb8-4b3a-4342-b03a-20c2d50dc2bf.mp4

File: .env.development.local
Comment (adamazad) [inline .env.development.local:1]: @berteotti does this solve #1019 for you?

File: src/components/swap/SwapButton.tsx
Comment (adamazad) [inline src/components/swap/SwapButton.tsx:34]: Did you mean `flex: 1 1`? I don't recall `1.1` being a valid value

**Review by adamazad — COMMENTED** [review]
Awesome work as always :)!

File: .env.development.local
Comment (berteotti) [reply .env.development.local:1]: It's perfect now! This solves it.

File: package.json
Comment (berteotti) [inline package.json:7]: Should we do it on pre-install as before? I didn't noticed this was removed and only ran `yarn codegen` after running `yarn install` and `yarn start` a couple of times

**Comment by wixzi** [comment]
The decision I came to in the case of `yarn codegen` is, we need to run it only once. We can add that step to our development docs. I am ok to add it in `pre-install` or `post-install`

May be I am wrong.

Why I didnt do that is it will run on each `yarn install`. I need to add `yarn install` as a pre step to `yarn react-scripts start`. Since we dont change the output of `yarn codegen` so often but we will change our packages so often. 

So if `yarn start` include yarn install as a pre-step its a good thing I feel. 

I am ok to rename it. I renamed to codegen so its easy to type if you are running it separate 😄 . 
I know we have autocomplete and history complete in our terminals.

File: package.json
Comment (wixzi) [reply package.json:7]: Since I added `yarn & react-scripts start`  you dont want to do `yarn install`. 

`yarn = yarn install`

I felt dont want to run it always 😞

File: src/components/swap/SwapButton.tsx
Comment (wixzi) [reply src/components/swap/SwapButton.tsx:34]: its like 110% flex can be a positive number 😄 so the other will be 90% or depending on the other elements flex

**Comment by wixzi** [comment]
@adamazad I will update codegen with scoping back. I will also fix this issues reported. Thank you for checking :)

File: src/state/swap/reducer.ts
Comment (wixzi) [reply src/state/swap/reducer.ts:111]: Now the implementation is a middle step since the eco-router was sync earlier. Once this is merged I can revisit and clean up the loading a bit. You dont need to update the loading since I am checking the output of In and Out  dexes length. 

Will use this loading once it is merged and it will be correct way to get the loading.

**Comment by wixzi** [comment]
> After-Effects_-box.mp4 
> This is my initial idea of how the loading would look like

Will try that animation as a next ticket. I was trying with css animations and moving up like that was little tricky. I can give another try as a next ticket.

**Comment by wixzi** [comment]
@adamazad can you review this again. 

@MilanVojnovic95 Can you test the code again. I have fixed the issues. I couldn't reproduce the issues now.

**Comment by MilanVojnovic95** [comment]
Retested and looks ok to me 👍

**Comment by velusgautam** [comment]
@adamazad @MilanVojnovic95 can you approve the pr if its good enough. 

Will add a ticket to work on loading again once adam merges the new hook with loading state

**Comment by adamazad** [comment]
@velusgautam seems good, I want to merge #977 as well :)

File: README.md
Comment (adamazad) [inline README.md:24]: Yarn doesn't have `ci`, 

```
yarn install --frozen-lockfile
```

**Review by adamazad — APPROVED** [review]

**Pull request merged**
